src/fields/CoordinatesInput/index.tsx

```diff
@@ -1,63 +1,98 @@
 
+import type { FieldsetProps } from '../../elements/Fieldset'
+import type { Coordinates } from '../../types'
 import type { Promisable } from 'type-fest'
 
-export type CoordinatesInputProps = {
+export type CoordinatesInputProps = FieldsetProps & {
   coordinatesFormat: CoordinatesFormat
-  defaultValue: number[]
+  defaultValue?: Coordinates | undefined
+  disabled?: boolean | undefined
+  error?: string | undefined
+  isLabelHidden?: boolean | undefined
   isLight?: boolean | undefined
-  onChange: (nextCoordinates: number[], coordinates: number[]) => Promisable<void>
+  label: string
+  onChange?:
+    | ((nextCoordinates: Coordinates | undefined, coordinates: Coordinates | undefined) => Promisable<void>)
+    | undefined
 }
 export function CoordinatesInput({
   coordinatesFormat,
   defaultValue,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  error,
+  isLabelHidden = false,
   isLight = false,
-  onChange
+  label,
+  onChange = noop,
+  ...nativeProps
 }: CoordinatesInputProps) {
+  const controlledError = useMemo(() => normalizeString(error), [error])
+  const hasError = useMemo(() => Boolean(controlledError), [controlledError])
+
   const getCoordinatesInput = useCallback(() => {
     switch (coordinatesFormat) {
       case CoordinatesFormat.DEGREES_MINUTES_SECONDS:
         return (
           <DMSCoordinatesInput
             coordinates={defaultValue}
             coordinatesFormat={CoordinatesFormat.DEGREES_MINUTES_SECONDS}
+            disabled={nativeProps.disabled}
             onChange={onChange}
           />
         )
+
       case CoordinatesFormat.DEGREES_MINUTES_DECIMALS:
         return (
           <DMDCoordinatesInput
             coordinates={defaultValue}
             coordinatesFormat={CoordinatesFormat.DEGREES_MINUTES_DECIMALS}
+            disabled={nativeProps.disabled}
             onChange={onChange}
           />
         )
+
       case CoordinatesFormat.DECIMAL_DEGREES:
-        return <DDCoordinatesInput coordinates={defaultValue as [number, number]} onChange={onChange} />
+        return (
+          <DDCoordinatesInput
+            coordinates={defaultValue as [number, number]}
+            disabled={nativeProps.disabled}
+            onChange={onChange}
+          />
+        )
+
       default:
         return undefined
     }
-  }, [defaultValue, onChange, coordinatesFormat])
+  }, [defaultValue, nativeProps.disabled, onChange, coordinatesFormat])
 
-  return <Box $isLight={isLight}>{getCoordinatesInput()}</Box>
-}
+  // TODO We must add a `handleDisable()` callback here to effectively empty the inputs when disabling this field.
+  useFieldUndefineEffect(nativeProps.disabled, onChange /* , handleDisable */)
 
-const Box = styled.div<{
-  $isLight: boolean
-}>`
-  color: ${p => p.theme.color.lightGray};
-  font-size: 13px;
-  text-align: left;
+  return (
+    <StyledFieldset isLegendHidden={isLabelHidden} isLight={isLight} legend={label} {...nativeProps}>
```

Comment:
I can find both `legend` and `label` in props, with `legend` overriding `label`, why ?

Reply:
On the published Fieldset component I use "legend" to explicitly explain that it's a `<fieldset>` with a `<legend>` within and not a <label>. A `<fieldset>` generally contains multiple inputs with their related `<label>` but it can (and should) only contain one `<legend>`.

Both labels and legends are used to find monitor-ui fields in the soon-to-be-integrated `cy.fill()` command (which is already how it works in monitorfish).